File: GradeTrackerAPI/Entities/Module.cs
namespace GradeTrackerAPI.Entities
{
    public class Module : BaseEntity
    {
        public string? Name { get; set; } = string.Empty;
        public bool? ShowOnDashboard { get; set; } = false;
        public double? AverageDesiredMark { get; set; }
        public int? CompetenceAreaId { get; set; }
        public CompetenceArea? CompetenceArea { get; set; }
        public int? TeacherId { get; set; }
        public Teacher? Teacher { get; set; }
        public int? UserId { get; set; }
        public User? User { get; set; }
        public ICollection<Mark>? Marks { get; set; }
    }
}

File: GradeTracker/src/app/_models/module.ts
import { User } from './user';
import { CompetenceArea } from './competenceArea';
import { MatTableDataSource } from "@angular/material/table";
import { Mark, IMark } from "./mark";
import { Teacher } from './teacher';

export class Module {
    id?: number;
    name?: string;
    competenceArea?: CompetenceArea;
    competenceAreaId? : number
    teacher?: Teacher;
    averageDesiredMark?: number;
    averageMark? : number;
    marks?: Mark[] | MatTableDataSource<Mark>;
    showOnDashboard?: boolean;
    userId? : number;
    isEditable? : boolean
}

export const ModuleColumns = [
  {
    key: 'name',
    type: 'text',
    label: 'Modul',
    required: true
  },
  {
    key: 'educationTypeId',
    type: 'number',
    label: 'Bildungstyp',
    required: true,
  },
  {
    key: 'competenceAreaId',
    type: 'number',
    label: 'Fachtyp',
    required: true,
  },
  {
    key: 'teacher',
    type: 'Teacher',
    label: 'Lehrer'
  },
  {
    key: 'isEditable',
    type: 'isEditable',
    label: ''
  }
]

export interface IModule {
  id?: number;
  name?: string;
  competenceArea?: CompetenceArea;
  competenceAreaId? : number
  teacher?: string;
  averageDesiredMark?: number;
  marks?: IMark[];
  showOnDashboard?: boolean;
}